sql.go

```diff
@@ -125,17 +173,61 @@ func (s *sql) UpdatePrefix(ctx context.Context, prefix Prefix) (Prefix, error) {
 	return prefix, tx.Commit()
 }
 
-func (s *sql) DeletePrefix(ctx context.Context, prefix Prefix) (Prefix, error) {
+func (s *sql) DeletePrefix(ctx context.Context, prefix Prefix, namespace string) (Prefix, error) {
+	if err := s.checkNamespaceExists(ctx, namespace); err != nil {
+		return Prefix{}, err
+	}
 	tx, err := s.db.Beginx()
 	if err != nil {
-		return Prefix{}, fmt.Errorf("unable to start transaction:%w", err)
+		return Prefix{}, fmt.Errorf("unable to start transaction: %w", err)
 	}
-	_, err = tx.ExecContext(ctx, "DELETE from prefixes WHERE cidr=$1", prefix.Cidr)
+	_, err = tx.ExecContext(ctx, "DELETE from prefixes_"+namespace+" WHERE cidr=$1", prefix.Cidr)
 	if err != nil {
-		return Prefix{}, fmt.Errorf("unable delete prefix:%w", err)
+		return Prefix{}, fmt.Errorf("unable delete prefix: %w", err)
 	}
 	return prefix, tx.Commit()
 }
 func (s *sql) Name() string {
 	return "postgres"
 }
+
+func (s *sql) CreateNamespace(ctx context.Context, namespace string) error {
+	if _, ok := s.tables.Load(namespace); !ok {
+		if _, err := s.db.ExecContext(ctx, createTableSQL(namespace)); err != nil {
```

Comment:
I think we should validate the namespace according the limitations of SQL for tablenames. All i found for postgres was this pointer: https://www.postgresql.org/docs/current/limits.html which says that identifiers must not exceed 63 bytes. Also only a small set of characters are allowed.

I mean if someone tries to create  a namespace which is not conformant with the SQL standard, an SQL Error is returned which is probably not so convenient.

Reply:
Funny you should say that, I ran into it yesterday. I'll add a check to make sure the namespace names are valid.

Reply:
Done